### PR TITLE
feat(chat): allow share a thread from Chat panel sidebar

### PR DESCRIPTION
**Pull request by liangfung**
## Changes
1. allow share a thread from Chat panel sidebar
2. implement client code view and navigation in Answer Engine

## Screen record
https://jam.dev/c/9194b48d-7335-4827-b188-ee832c265c33
image

### CODE REVIEW

**Comment by wsxiaoys** [comment]
LGTM for `ee/tabby-schema/src/schema/mod.rs`

File: ee/tabby-ui/app/chat/page.tsx [hunk 1]
```diff
@@ -263,6 +263,23 @@ export default function ChatPage() {
     server?.navigate(context, opts)
   }
 
+  const onOpenExternal = useMemo(() => {
+    // FIXME check capability
+    const hasCapability = true
```
Comment (Sma1lboy) [inline ee/tabby-ui/app/chat/page.tsx:268]: Now we can use the `hasCapability` check from the client API.

**Review by Sma1lboy — CHANGES_REQUESTED** [review]
Otherwise, LGTM

File: clients/tabby-chat-panel/src/index.ts [hunk 2]
```diff
@@ -77,6 +77,7 @@ export interface ClientApiMethods {
 
   onKeyboardEvent: (type: 'keydown' | 'keyup' | 'keypress', event: KeyboardEventInit) => void
 
+  onOpenExternal: (url: string) => Promise<boolean>
```
Comment (wsxiaoys) [inline clients/tabby-chat-panel/src/index.ts:80]:
```suggestion
  onOpenExternal?: (url: string) => Promise<boolean>
```

**Review by Sma1lboy — APPROVED** [review]
LGTM

**Pull request merged**